src/modules/mana/sagas.ts

```diff
@@ -346,12 +354,27 @@ function* handleFinishWithdrawalRequest(action: FinishWithdrawalRequestAction) {
     const matic = new MaticPOSClient(config)
 
     const tx: { transactionHash: string } = yield call(() =>
-      matic.exitERC20(withdrawal.hash, { from })
+      matic.exitERC20(withdrawal.initializeHash, {
+        from,
+        onTransactionHash: (hash: string) => {
+          store.dispatch(setWithdrawalFinalizeHash(withdrawal, hash))
+        },
+      })
     )
 
-    yield put(finishWithdrawalSuccess(withdrawal, chainId, tx.transactionHash))
+    const storeWithdrawal: Withdrawal = yield getStoreWithdrawalByHash(
+      withdrawal.initializeHash
+    )
```

Comment:
Asing the store for the withdrawal object because the one that is already in scope is not the one updated with the finalizeHash